File: app/src/main/java/com/python/companion/db/typeconverters/InstantConverter.java
package com.python.companion.db.typeconverters;

import androidx.room.TypeConverter;
import java.time.Instant;
public class InstantConverter {
    @TypeConverter
    public static Instant dateFromTimestamp(Long value) {
        return value == null ? null : Instant.ofEpochSecond(value);
    }

    @TypeConverter
    public static Long dateToTimestamp(Instant moment) {
        return moment == null ? null : moment.getEpochSecond();
    }
}

File: db/src/main/java/org/python/db/typeconverters/InstantConverter.kt
package org.python.db.typeconverters

import androidx.room.TypeConverter
import java.time.Instant

object InstantConverter {
    @TypeConverter
    fun dateFromTimestamp(value: Long?): Instant? {
        return if (value == null) null else Instant.ofEpochSecond(value)
    }

    @TypeConverter
    fun dateToTimestamp(moment: Instant?): Long? {
        return moment?.epochSecond
    }
}